src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.String.cs

```diff
@@ -9,6 +9,353 @@ namespace System.Text.Json
+            {
+                WriteStringByOptionsPropertyName(propertyName);
+            }
+            _currentDepth &= JsonConstants.RemoveFlagsBitMask;
+            _tokenType = JsonTokenType.PropertyName;
+        }
+
+        private void WriteStringEscapeProperty(ReadOnlySpan<char> propertyName, int firstEscapeIndexProp)
+        {
+            Debug.Assert(int.MaxValue / JsonConstants.MaxExpansionFactorWhileEscaping >= propertyName.Length);
+
+            char[] propertyArray = null;
+
+            if (firstEscapeIndexProp != -1)
+            {
+                int length = JsonWriterHelper.GetMaxEscapedLength(propertyName.Length, firstEscapeIndexProp);
+
+                Span<char> escapedPropertyName;
+                if (length > JsonConstants.StackallocThreshold)
+                {
+                    propertyArray = ArrayPool<char>.Shared.Rent(length);
+                    escapedPropertyName = propertyArray;
+                }
+                else
+                {
+                    // Cannot create a span directly since it gets assigned to parameter and passed down.
+                    unsafe
+                    {
+                        char* ptr = stackalloc char[length];
+                        escapedPropertyName = new Span<char>(ptr, length);
+                    }
+                }
+
+                JsonWriterHelper.EscapeString(propertyName, escapedPropertyName, firstEscapeIndexProp, out int written);
+                propertyName = escapedPropertyName.Slice(0, written);
+            }
+
+            WriteStringByOptionsPropertyName(propertyName);
+
+            if (propertyArray != null)
+            {
+                ArrayPool<char>.Shared.Return(propertyArray);
+            }
+        }
+
+        private void WriteStringByOptionsPropertyName(ReadOnlySpan<char> propertyName)
+        {
+            ValidateWritingProperty();
+            if (Options.Indented)
+            {
+                WriteStringIndentedPropertyName(propertyName);
+            }
+            else
+            {
+                WriteStringMinimizedPropertyName(propertyName);
+            }
+        }
+
+        private void WriteStringMinimizedPropertyName(ReadOnlySpan<char> escapedPropertyName)
+        {
+            Debug.Assert(escapedPropertyName.Length <= JsonConstants.MaxTokenSize);
+            Debug.Assert(escapedPropertyName.Length < (int.MaxValue - 4) / JsonConstants.MaxExpansionFactorWhileTranscoding);
+
+            // All ASCII, 2 quotes for property name, and 1 colon => escapedPropertyName.Length + 3
+            // Optionally, 1 list separator, and up to 3x growth when transcoding
+            int maxRequired = (escapedPropertyName.Length * JsonConstants.MaxExpansionFactorWhileTranscoding) + 4;
+
+            if (_memory.Length - BytesPending < maxRequired)
+            {
+                Grow(maxRequired);
+            }
+
+            Span<byte> output = _memory.Span;
+
+            if (_currentDepth < 0)
```

Comment:
I had to go find the comment on _currentDepth to understand how a depth could be negative.  Consider instead creating a private property that does this check and is appropriately named to describe what it's actually checking.

Reply:
Copy-pasted from existing code, so will have to fix this across the board. Out-of-scope for this PR.